### PR TITLE
Add test suite and clean up package internals.

### PR DESCRIPTION
**Pull request by thinkverse**
This PR includes a test case, a `.gitignore` file to prevent `vendor` and other files from being tracked.

`.gitattributes` was added to prevent the `tests` directory and other select files from getting downloaded when the package is required. This saves both bandwidth and storage space for users.

### CODE REVIEW

**Review by deployHuman — COMMENTED** [review]
The 0 is below 1 so it would have defaulted to the 10 ^^ 
or am i missing something ?

**Review by deployHuman — APPROVED** [review]
i hade .ignore on my own file cuz i have lots of other stuff there, but yes ofc it should be here :)

**Review by deployHuman — APPROVED** [review]

**Comment by thinkverse** [comment]
> The 0 is below 1 so it would have defaulted to the 10 ^^
> or am i missing something ?

Previously it checked if `$length` was **less than** `0`, this allowed a value of `0` to be passed in, causing the fatal error to be thrown. As shown in this image where I reverted to the earlier check and the test fails.

Screen Shot 2022-02-19 at 6 54 09 PM

File: src/Helper.php [hunk 1]
```diff
@@ -2,23 +2,19 @@
...
-     * Returns a random string of a specified byte size
+     * Returns a random string of a specified byte size,
+     * note that 1 byte contains 2 characters.
      * 
      * @param int $length in bytes
      * @return string $key
      */
     public static function getRandomKey(int $length): string
     {
-        if ($length < 1) {
```
Comment (deployHuman) [inline src/Helper.php:18]: it was like this before ;) but sure

Comment (thinkverse) [reply src/Helper.php:18]: It was my mistake there, mine was less than `0` for some reason even though the source is less than `1`. Must've changed it when I removed the redundant `$key` variable. 🤷‍♂️

At least the test case supports that `getRandomKey` reverts to a default key length. 😂

Comment (deployHuman) [reply src/Helper.php:18]: haha this is why OpenSource is good, more then one pair of eyes! :) your edit is a nice refactor to skip the variable thou! like it

Comment (thinkverse) [reply src/Helper.php:18]: Changes the PR title since the original intent didn't really match now. 👍

**Pull request merged**